apps/src/maze/harvesterBlocks.js

```diff
@@ -0,0 +1,231 @@
+    var doCode = generator.statementToCode(this, 'DO');
+    var elseCode = generator.statementToCode(this, 'ELSE');
+    var code = `if (${argument}) {\n${doCode}} else {\n${elseCode}}\n`;
+    return code;
+  };
+}
+
+function addIfSpecificCropHasBlock(blockly, generator, crop) {
+  blockly.Blocks[`harvester_ifHas${capitalizeFirstLetter(crop)}`] = {
+    helpUrl: '',
+    init: function () {
+      this.setHSV(196, 1.0, 0.79);
+      this.appendDummyInput()
+          .appendTitle([msg.ifCode(), msg[`has${crop}`]()].join(' '));
+      this.setInputsInline(true);
+      this.appendStatementInput('DO')
+          .appendTitle(msg.doCode());
+      this.setPreviousStatement(true);
+      this.setNextStatement(true);
+    }
+  };
+
+  generator[`harvester_ifHas${capitalizeFirstLetter(crop)}`] = function () {
+    var argument = `Maze.has${capitalizeFirstLetter(crop)}('block_id_${this.id}')`;
+    var branch = generator.statementToCode(this, 'DO');
+    var code = `if (${argument}) {\n${branch}}\n`;
+    return code;
+  };
+}
+
+function addWhileSpecificCropHasBlock(blockly, generator, crop) {
+  blockly.Blocks[`harvester_whileHas${capitalizeFirstLetter(crop)}`] = {
+    helpUrl: '',
+    init: function () {
+      this.setHSV(322, 0.90, 0.95);
+      this.appendDummyInput()
+          .appendTitle([msg.whileMsg(), msg[`has${crop}`]()].join(' '));
+      this.setInputsInline(true);
+      this.appendStatementInput('DO')
+          .appendTitle(msg.doCode());
+      this.setPreviousStatement(true);
+      this.setNextStatement(true);
+    }
+  };
+
+  generator[`harvester_whileHas${capitalizeFirstLetter(crop)}`] = function () {
+    var argument = `Maze.has${capitalizeFirstLetter(crop)}('block_id_${this.id}')`;
+    var branch = generator.statementToCode(this, 'DO');
+    var code = `while (${argument}) {\n${branch}}\n`;
+    return code;
+  };
+}
+
+exports.install = function (blockly, blockInstallOptions) {
+  var skin = blockInstallOptions.skin;
+  var isK1 = blockInstallOptions.isK1;
+
+  var generator = blockly.Generator.get('JavaScript');
+  blockly.JavaScript = generator;
+
+  CROPS.forEach(crop => {
+    // harvest crop
+    blockUtils.generateSimpleBlock(blockly, generator, {
+      name: `harvester_${crop}`,
+      helpUrl: '',
+      title: isK1 ? msg.pick() : `${msg.pick()} ${msg[crop]()}`,
+      titleImage: isK1 ? skin[crop] : undefined,
+      tooltip: msg[`${crop}Tooltip`](),
+      functionName: `Maze.get${capitalizeFirstLetter(crop)}`
+    });
+
+    addIfAtSpecificCropBlock(blockly, generator, crop);
+    addIfAtSpecificCropElseBlock(blockly, generator, crop);
+    addIfSpecificCropHasBlock(blockly, generator, crop);
+    addWhileSpecificCropHasBlock(blockly, generator, crop);
+  });
+
+  const AT_OPTIONS = CROPS.map(crop => [msg[crop](), capitalizeFirstLetter(crop)]);
+  const HAS_OPTIONS = CROPS.map(crop => [msg[`has${crop}`](), capitalizeFirstLetter(crop)]);
```

Comment:
because `"has <crop>"` isn't actually `"has <crop>"`, but rather something more complicated; specifically, one of "there are beans", "there is corn", or "there are pumpkins",